BaseStation/MainWindow/VideoStreamer/RTSPVideoWindow.xaml.cs

```diff
@@ -0,0 +1,121 @@
+using System.Windows.Input;
+using System.Windows.Interop;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+
+namespace HuskyRobotics.UI.VideoStreamer
+{
+    /// <summary>
+    /// Interaction logic for RTSPVideoWindow.xaml
+    /// </summary>
+    /// 
+    // gst-launch-1.0 rtspsrc -v location = rtsp://192.168.0.42 user-id=admin user-pw=1234 latency=0 ! rtpjpegdepay ! jpegdec ! d3dvideosink
+    public partial class RTSPVideoWindow : Window, VideoWindow
+    {
+        private Pipeline Pipeline;
+
+        private Bin RTSP = (Bin) ElementFactory.Make("rtspsrc");
+        private Element Depay = ElementFactory.Make("rtpjpegdepay");
+        private Element Dec = ElementFactory.Make("jpegdec");
+        private Element VideoSink = ElementFactory.Make("d3dvideosink");
+
+        private int Port;
+        private int BufferSizeMs;
+        private string RecordingPath;
+        private string _streamName;
+
+        public string StreamName
+        {
+            get { return this._streamName; }
+            set
+            {
+                this.Title = value;
+                this._streamName = value;
+            }
+        }
+
+        public RTSPVideoWindow(int Port, string StreamName, string RecordingPath, int BufferSizeMs = 200)
+        {
+            DataContext = this;
+            this.Port = Port;
+            this.StreamName = StreamName;
+            this.BufferSizeMs = BufferSizeMs;
+            this.RecordingPath = RecordingPath;
+
+            Pipeline = new Pipeline();
+            
+
+            Window window = GetWindow(this);
+            WindowInteropHelper wih = new WindowInteropHelper(this);
+            wih.EnsureHandle(); // Generate Window Handle if current HWND is NULL (always occurs because background window)
+            VideoOverlayAdapter overlay = new VideoOverlayAdapter(VideoSink.Handle);
+            overlay.WindowHandle = wih.Handle;
+
+            Pipeline["message-forward"] = true;
+            RTSP["location"] = "rtsp://" + StreamName;
+            RTSP["user-id"] = "admin";
+            RTSP["user-pw"] = "1234";
+            RTSP["latency"] = BufferSizeMs;
+
+
+            Pipeline.Add(RTSP, Depay, Dec, VideoSink);
+            RTSP.Link(Depay);
+            RTSP.PadAdded += RTSPPadAdded;
+            Depay.Link(Dec);
+            Dec.Link(VideoSink);
+
+
+            Pipeline.SetState(State.Null);
+
+            Closing += OnCloseEvent;
+            InitializeComponent();
+        }
+
+        private void RTSPPadAdded(object o, PadAddedArgs args)
+        {
+            Pad Sink = Depay.GetStaticPad("sink");
+            args.NewPad.Link(Sink);
+        }
+
+        /// <summary>
+        /// Window (this) must be visible before this method is called.
+        /// </summary>
+        public void StartStream()
+        {
+            StateChangeReturn s = Pipeline.SetState(State.Playing);
+        }
+
+        private string GetFilename()
```

Comment:
What is this used for? What filename is this referring to? The name should answer these questions.